components/cam/src/physics/cam/micro_p3.F90

```diff
@@ -468,12 +468,12 @@ SUBROUTINE p3_main_pre_main_loop(kts, kte, kbot, ktop, kdir, log_predictNc, dt,
   END SUBROUTINE p3_main_pre_main_loop
 
   SUBROUTINE p3_main_main_loop(kts, kte, kbot, ktop, kdir, log_predictNc, dt, odt, &
-       pres, pdel, dzq, npccn, exner, inv_exner, inv_lcldm, inv_icldm, inv_rcldm, naai, qc_relvar, icldm, lcldm, rcldm,&
-       t, rho, inv_rho, qvs, qvi, sup, supi, rhofacr, rhofaci, acn, qv, th, qc, nc, qr, nr, qitot, nitot, &
+       pres, pdel, dzq, ncnuc, exner, inv_exner, inv_lcldm, inv_icldm, inv_rcldm, naai, qc_relvar, icldm, lcldm, rcldm,&
+       t, rho, inv_rho, qvs, qvi, supi, rhofacr, rhofaci, acn, qv, th, qc, nc, qr, nr, qitot, nitot, &
        qirim, birim, xxlv, xxls, xlf, qc_incld, qr_incld, qitot_incld, qirim_incld, nc_incld, nr_incld, &
        nitot_incld, birim_incld, mu_c, nu, lamc, cdist, cdist1, cdistr, mu_r, lamr, logn0r, cmeiout, prain, &
        nevapr, prer_evap, vap_cld_exchange, vap_liq_exchange, vap_ice_exchange, liq_ice_exchange, pratot, &
-       prctot, p3_tend_out, log_hydrometeorsPresent)
+       prctot, p3_tend_out, log_hydrometeorsPresent, qc_old, nc_old, qr_old, nr_old, qitot_old, nitot_old, qv_old, th_old)
```

Comment:
Kobby - I don't understand why you need to add these variables to the main_main_loop interface. The code you're modifying just assigned p3_tend_out to values *within* main_main_loop. Why can't you do that, but with locally defined variables?

Reply:
@PeterCaldwell 

We talked about it in the previous PR.
The tendencies are initialized with in-cloud value, but at the end of P3 the output tendencies are calculate based on cell-averaged values.

Reply:
Hmm. I still don't understand. To my mind, your *_old variables are exact replacements for the p3_tend_out assignments around line 616 of the code before your changes. Those p3_tend_out variables sure look like cell-average values to me. Also, those p3_tend_out variables don't seem to be causing any trouble by having their assignment be local. Then around L964 of the old code, those p3_tend_out variables are being used.

Is the issue that you're trying to use the state before aerosol activation in your p3_tend_out tendency for nc? I'm still confused...

Reply:
Peter, please look inside `p3_main_pre_main_loop`, specifically at the routine `calculate_incloud_mixingratios`.
That means all the required fields for initialization are not cell-averaged, but in-cloud values.

Reply:
Hi Kobby - I'm still totally confused. calculate_incloud_mixingratios takes qc as input and returns qc_incld as output. p3_tend_out for qc tendency is (qc - qc_old)/dt... which uses cell-average values for everything. I think instead of having in-cloud versus cell average problems, you wanted *_old to be defined such that pre_main_loop and main_main_loop changes are both included??? If that's the case, it would probably be better to calculate those p3_tend_out tendencies that require *_old from p3_main rather than half in p3_main and half in p3_main_main_loop. Does this make sense?

Reply:
Yes, OK, will move the p3_tend_out that involves cell_average *_old to p3_main.

Reply:
Ok, great. Thanks!